// terminal-styling/dist/modules/console.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Console = void 0;
var chalk = require('chalk');
var colors_1 = require("./colors");
var Console = /** @class */ (function () {
    function Console() {
        this.err = function (consoleLogMessage) {
            console.error(colors_1.red(consoleLogMessage));
        };
        this.warn = function (consoleLogMessage) {
            console.warn(colors_1.yellow(consoleLogMessage));
        };
        this.info = function (consoleLogMessage) {
            console.info(colors_1.green(consoleLogMessage));
        };
        this.table = function () {
            var args = [];
            for (var _i = 0; _i < arguments.length; _i++) {
                args[_i] = arguments[_i];
            }
            console.table.apply(console, args);
        };
        this.customColor = function (text_hex, consoleLogMessage, is_bold, bg_hex) {
            var color;
            if (bg_hex !== undefined) {
                color = chalk.hex(text_hex).bgHex(bg_hex);
                if (is_bold === true) {
                    color = chalk.hex(text_hex).bgHex(bg_hex).bold;
                }
                else {
                    color = chalk.hex(text_hex).bgHex(bg_hex);
                }
            }
            else {
                color = chalk.hex(text_hex);
                if (is_bold === true) {
                    color = chalk.hex(text_hex).bold;
                }
                else {
                    color = chalk.hex(text_hex);
                }
            }
            console.log(color(consoleLogMessage));
        };
    }
    return Console;
}());
exports.Console = Console;
